File: app/src/main/java/com/lumbralessoftware/voterussia2018/player/PlayerListContract.java
package com.lumbralessoftware.voterussia2018.player;

import com.lumbralessoftware.voterussia2018.BaseView;
import com.lumbralessoftware.voterussia2018.Player;

import java.util.List;

/**
 * Created by javiergonzalezcabezas on 21/5/18.
 */

public class PlayerListContract {

    interface Presenter {
        void fetch();
    }

    interface View extends BaseView<Presenter> {
        void showPlayer(List<Player> list);

        void showError();

        void setLoadingIndicator(boolean active);

    }
}

File: app/src/main/java/com/lumbralessoftware/voterussia2018/player/PlayerListContract.kt
package com.lumbralessoftware.voterussia2018.player

import com.lumbralessoftware.voterussia2018.BaseView
import com.lumbralessoftware.voterussia2018.NewPlayer

/**
 * Created by javiergonzalezcabezas on 31/5/18.
 */
interface PlayerListContract {

    interface Presenter {
        fun fetch()

        fun fetchPlayerWithPosition(position :Int)

        fun fetchPlayerWithTeam(team :Int)

        fun goToRating(id: Int, name: String, image:String)
    }

    interface View : BaseView<Presenter> {
        fun showPlayer(list: List<NewPlayer>)

        fun showError()

        fun setLoadingIndicator(active: Boolean)

        fun noInternet()

    }
}